File: Python/Path.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

import os

pt=os.path.normpath(os.path.join(os.getcwd(),__file__))

print(f"""
ホームディレクトリ:       {os.path.expanduser("~")}
カレントディレクトリ:     {os.getcwd()}

このファイルのフルパス:   {os.path.abspath(pt)}
このファイルの実行パス:   {__file__}
このファイルの名前:       {os.path.basename(pt)}
このファイルのある場所:   {os.path.dirname(pt)}
""")

print("\r\n")

File: Perl/Path.pl
#! /usr/bin/env perl

use Cwd;
use File::Basename; # dirname用

print <<"Path";
ホームディレクトリ:       @{[$ENV{"HOME"}]}
カレントディレクトリ:     @{[Cwd::getcwd()]}

このファイルのフルパス:   @{[Cwd::realpath($0)]}
このファイルの実行パス:   $0
このファイルの名前:       @{[basename($0)]}
このファイルのある場所:   @{[Cwd::realpath(dirname($0))]}
Path

print "\r\n\r\n";